src/pages/Login.tsx

```diff
@@ -91,54 +91,71 @@ const validatePassword = (v: string) => {
   return v.length >= 8 ? '' : PW_TOO_SHORT;
 };
 
+export const UserInfoContext = createContext<{ id: string; pw: string } | null>(
+  null,
+);
+
 function Login() {
   const navigate = useNavigate();
   const location = useLocation();
   const from = location.state?.from?.pathname || '/';
 
   const id = useInput(validateEmail);
   const pw = useInput(validatePassword);
+  const [userInfo, setUserInfo] = useState<{ id: string; pw: string } | null>(
+    null,
+  );
 
   const isFormValid = id.isValid && pw.isValid;
 
   const handleLogin = (e: React.MouseEvent<HTMLButtonElement>) => {
     e.preventDefault();
     if (!isFormValid) return;
+
+    const nextUser = { id: id.value, pw: pw.value };
+
+    setUserInfo(nextUser);
+
+    sessionStorage.setItem('userId', nextUser.id);
+    // sessionStorage.setItem('userPw',`${userInfo?.pw}`);
+
     navigate(from, { replace: true });
   };
```

Comment:
이 일련의 로그인 과정도 useUser라는 커스텀 훅에서 추상화해주고, 컴포넌트에서는 추상화된 함수를 꺼내쓰면 좋을 것 같아요.

```suggestion
  const { user, login } = useUser();

  const handleLogin = (e: React.MouseEvent<HTMLButtonElement>) => {
    e.preventDefault();
    if (!isFormValid) return;

    login({ id: id.value, pw: pw.value });

    navigate(from, { replace: true });
  };
```